File: frontend/src/types/server_operation.ts
export type ServerOpStatus =
  | "pending_review"
  | "executing"
  | "completed"
  | "failed"
  | "rejected";

export interface ServerOperation {
  id: string;
  tenant_id: string;
  user_id: string;
  server_id: string;
  description: string;
  commands: string[];
  output: string | null;
  status: ServerOpStatus;
  error_message: string | null;
  review_comment: string | null;
  reviewer_id: string | null;
  reviewed_at: string | null;
  requester_name: string | null;
  requester_email: string | null;
  server_name: string | null;
  server_host: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/server_operation.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.server_operation import ServerOpStatus


class ExecRequest(BaseModel):
    description: str = Field(..., min_length=1, max_length=500)
    commands: list[str] = Field(..., min_length=1)


class ReviewRequest(BaseModel):
    approved: bool
    comment: str = ""


class ServerOperationRead(BaseModel):
    id: UUID
    tenant_id: UUID
    user_id: UUID
    server_id: UUID
    description: str
    commands: list[str]
    output: str | None
    status: ServerOpStatus
    error_message: str | None
    review_comment: str | None
    reviewer_id: UUID | None
    reviewed_at: datetime | None
    requester_name: str | None
    requester_email: str | None
    server_name: str | None
    server_host: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
